neutron/services/loadbalancer/drivers/haproxy/namespace_driver.py

```diff
@@ -362,7 +397,8 @@ def remove_orphans(self, known_pool_ids):
         orphans = (pool_id for pool_id in os.listdir(self.state_path)
                    if pool_id not in known_pool_ids)
         for pool_id in orphans:
-            if self.exists(pool_id):
+            socket_path = self._get_state_file_path(pool_id, 'sock', False)
+            if os.path.exists(socket_path):
```

Comment:
这里有种情况就是，在agent不知道pool id对应的port id的时候， 去server端获取也有可能获取不到对应的port
id（例如这个pool id在server的数据库中已经被删除了）

2016-06-16 14:01 GMT+08:00 huntxu <notifications@github.com>:

> In neutron/services/loadbalancer/drivers/haproxy/namespace_driver.py
> <https://github.com/eayunstack/neutron/pull/34#discussion_r67291056>:
>
> > @@ -362,7 +397,8 @@ def remove_orphans(self, known_pool_ids):
> >          orphans = (pool_id for pool_id in os.listdir(self.state_path)
> >                     if pool_id not in known_pool_ids)
> >          for pool_id in orphans:
> > -            if self.exists(pool_id):
> > +            socket_path = self._get_state_file_path(pool_id, 'sock', False)
> > +            if os.path.exists(socket_path):
>
> 我的意思是去修改这里那个exists()函数，一样是让他通过pool_id来判断进程是否存在就可以，没有必要每个调用self.exists都修改一次。
> 如果有orphan
> pool存在，那么意味着agent可能也不知道它相对应的namespace存在与否的情况，不管是直接这样处理还是判断exists()的返回值然后进去undeploy_instance，进程会被杀掉，但是对应的namespace还会存在，这也是个问题。所以应该考虑在undeploy_instance中当agent无法使用pool_id去找到对应的port_id时，去server处获取，才能完整解决。
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/neutron/pull/34/files/0a12962b43949b28db67d0068623c19f9cf64c55#r67291056>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGUT0WFKNItNr7RyXmPKC4lnLSeDNyS9ks5qMOa0gaJpZM4I2I7z>
> .
>